pzsd_bot/__main__.py

```diff
@@ -1,48 +1,214 @@
+        recipient_name = match["recipient_name"]
+        recipient_id = match["recipient_id"]
+
+        async with engine.connect() as conn:
+            result = await conn.execute(
+                select(pzsd_user).where(
+                    pzsd_user.c.discord_snowflake == str(message.author.id)
+                )
+            )
+            try:
+                bestower = result.one_or_none()
+            except MultipleResultsFound:
+                logger.error(
+                    "Multiple users with snowflake '%s' found in user table",
+                    message.author.id,
+                )
+                raise
+
+            if bestower is None:
+                logger.info(
+                    "User '%s' with snowflake '%s' tried to bestow points but wasn't in the user table",
+                    message.author.name,
+                    message.author.id,
+                )
+                return
+
+            if recipient_name:
+                result = await conn.execute(
+                    select(pzsd_user).where(pzsd_user.c.name == recipient_name.lower())
+                )
+            else:
+                result = await conn.execute(
+                    select(pzsd_user).where(
+                        pzsd_user.c.discord_snowflake == recipient_id
+                    )
+                )
+            try:
+                recipient = result.one_or_none()
+            except MultipleResultsFound:
+                recipient_type = "name" if recipient_name else "snowflake"
+                logger.error(
+                    "Multiple users with %s '%s' found in user table",
+                    recipient_type,
+                    recipient_name or recipient_id,
+                )
+                raise
+
+            if recipient is None:
+                logger.info(
+                    "%s tried to bestow points to '%s' but they weren't in the user table",
+                    bestower.name,
+                    recipient_name or recipient_id,
+                )
+                return
+
+        excessive_point_violation = abs(point_amount) > POINT_MAX_VALUE
+        if excessive_point_violation:
+            logger.info(
+                "%s tried to give %s more than the max allowed points (%s)",
+                bestower.name,
+                recipient.name,
+                format(point_amount, ','),
+            )
+            return
+
+        self_point_violation = bestower.id == recipient.id
+        if self_point_violation:
+            logger.info(
+                "%s attempted to give themselves %s points. Very naughty.",
+                bestower.name,
+                point_amount,
+            )
+        else:
+            logger.info(
+                "%s awarding %s point(s) to %s",
+                bestower.name,
+                format(point_amount, ','),
+                recipient.name,
+            )
+
+        if not self_point_violation:
```

Comment:
There should only be one check of if self_point_voilation, refactor this into the previous if